File: chatbot.js
function sendMessage() {

    const input = document.getElementById("userInput");
    const chatBox = document.getElementById("chatBox");

    const message = input.value.trim().toLowerCase();

    if (message === "") return;

    chatBox.innerHTML += `
        <p><strong>You:</strong> ${input.value}</p>
    `;

    let reply = "";

    if (
        message.includes("hi") ||
        message.includes("hello") ||
        message.includes("hii")
    ) {

        reply =
        "👋 Hello! I am CareerX AI. Ask me about careers, skills, roadmaps, or companies.";

    } else if (
        message.includes("career") ||
        message.includes("careers")
    ) {

        reply =
        "🚀 Popular careers: AI Engineer, Software Developer, Data Scientist, Mobile App Developer, Cloud Engineer, Cyber Security Analyst.";

    } else if (
        message.includes("company") ||
        message.includes("companies") ||
        message.includes("apply")
    ) {

        reply =
        "🏢 Top companies: Google, Microsoft, Amazon, Zoho, Infosys, TCS, Wipro, Accenture, Cognizant.";

    } else if (
        message.includes("ai engineer")
    ) {

        reply =
        "🤖 Learn Python → Machine Learning → Deep Learning → TensorFlow → Build AI Projects.";

    } else if (
        message.includes("software developer")
    ) {

        reply =
        "💻 Learn Java, Python, SQL, Git and Data Structures.";

    } else if (
        message.includes("mobile app")
    ) {

        reply =
        "📱 Learn Dart, Flutter, Firebase and build Android applications.";

    } else if (
        message.includes("roadmap")
    ) {

        reply =
        "🗺️ Choose a career role first. Then I can suggest a roadmap.";

    } else if (
        message.includes("skills")
    ) {

        reply =
        "🎯 Popular skills: Python, Java, SQL, Git, Flutter, React, AWS, Machine Learning.";

    } else {

        reply =
        "❓ I didn't understand that. Ask about careers, skills, roadmaps or companies.";
    }

    chatBox.innerHTML += `
        <p><strong>CareerX AI:</strong> ${reply}</p>
    `;

    chatBox.scrollTop = chatBox.scrollHeight;

    input.value = "";
}

File: chatbot.py
def career_chatbot(message, resume_data):


    message = message.lower()


    skills = resume_data.get(
        "skills",
        []
    )


    role = resume_data.get(
        "target_role",
        "Software Developer"
    )


    companies = resume_data.get(
        "companies",
        []
    )


    courses = resume_data.get(
        "courses",
        []
    )


    roadmap = resume_data.get(
        "roadmap",
        []
    )


    missing = resume_data.get(
        "skill_gap",
        {}
    ).get(
        "missing_skills",
        []
    )



    # Career Question

    if "career" in message or "role" in message:

        return f"""

Based on your resume:

Recommended Role:
{role}

Your Skills:
{', '.join(skills)}

You should focus on improving:
{', '.join(missing)}

"""



    # Company Question

    elif "company" in message or "job" in message:


        if companies:

            company_names = [
                c["name"]
                for c in companies
            ]


            return f"""

Recommended companies for you:

{', '.join(company_names)}

Apply for roles matching:
{role}

"""


    # Course Question

    elif "course" in message or "learn" in message:


        if courses:

            course_names = [
                c["course"]
                for c in courses
            ]


            return f"""

Recommended learning:

{', '.join(course_names)}

These will help you close your skill gap.

"""



    # Skill Question

    elif "skill" in message:


        return f"""

Current Skills:

{', '.join(skills)}


Missing Skills:

{', '.join(missing)}

"""


    # Roadmap

    elif "roadmap" in message or "plan" in message:


        return "\n".join(roadmap)



    else:


        return """

I can help you with:

• Career recommendation
• Skill improvement
• Company suggestions
• Learning courses
• Career roadmap

Ask me anything about your career.

"""